tools/clang/test/CodeGenHLSL/batch/expressions/conversions_and_casts/array_copy.hlsl

```diff
@@ -0,0 +1,32 @@
+// RUN: %dxc -E main -T ps_6_0 %s | FileCheck %s
+
+// Make sure memcpy float3[1], float3[1][1] works.
```

Comment:
Can you further reduce this repro? The bug is certainly more general than float3[1] to float3[1][1], what about with scalars? what if the other array is doubly nested? what about structs instead of arrays?

This is fine, but we want to know how much more there is to this bug.